src/python/bezier/hazmat/clipping.py

```diff
@@ -235,6 +314,76 @@ def clip_range(nodes1, nodes2):
        is meant to be used to find tangent intersections for already
        subdivided (i.e. sufficiently zoomed in) curve segments.
 
+    Two B |eacute| zier curves :math:`B_1(s)` and :math:`B_2(t)` are defined by
+    ``nodes1`` and ``nodes2``. The "fat line" (see :func:`compute_fat_line`)
+    for :math:`B_1(s)` is used to narrow the range of possible :math:`t`-values
+    in an intersection by considering the distance polynomial for
+    :math:`B_2(t)`:
+
+    .. math::
+
+       d(t) = \sum_{j = 0}^m \binom{n}{j} t^j (1 - t)^{n - j} \cdot d_j
+
+    Here :math:`d_j = a x_j + b y_j + c` are the distances of each control
+    point :math:`(x_j, y_j)` of :math:`B_2(t)` to the implicit line for
+    :math:`B_1(s)`.
+
+    Consider the following pair of B |eacute| zier curves and the distances
+    from **all** of the control points to the implicit line for :math:`B_1(s)`:
+
+    .. testsetup:: clip-range-start, clip-range
+
+       import numpy as np
+       import bezier
+       from bezier.hazmat.clipping import clip_range
+
+       nodes1 = np.asfortranarray([
+           [2.0, 4.5, 2.5, 5.0],
+           [0.0, 1.0, 3.0, 4.0],
+       ])
+       nodes2 = np.asfortranarray([
+           [-0.25 , 3.75 , 7.0  ],
+           [ 3.125, 0.875, 3.125],
+       ])
+
+    .. doctest:: clip-range-start
+       :options: +NORMALIZE_WHITESPACE
+
+       >>> nodes1
+       array([[2. , 4.5, 2.5, 5. ],
+              [0. , 1. , 3. , 4. ]])
+       >>> nodes2
+       array([[-0.25 ,  3.75 ,  7.   ],
+              [ 3.125,  0.875,  3.125]])
+
+    .. image:: ../../images/clip_range.png
+       :align: center
+
+    The distances from the control points of :math:`B_2(t)` define the
+    distance polynomial :math:`d(t)`. By writing this polynomial as a
+    B |eacute| zier curve, a convex hull can be formed. The intersection of
+    this convex hull with the "fat line" of :math:`B_1(s)` determines the
+    extreme :math:`t` values possible and allows clipping the range of
+    :math:`B_2(t)`:
+
+    .. image:: ../../images/clip_range_distances.png
+       :align: center
+
+    .. doctest:: clip-range
+       :options: +NORMALIZE_WHITESPACE
+
+       >>> s_min, s_max = clip_range(nodes1, nodes2)
+       >>> s_min
+       0.25
+       >>> np.allclose(s_max, 0.875, rtol=0.5 ** 52, atol=0.0)
```

Comment:
I'm less than happy about the fact that the value returned is `0.8749999999999999 == 0x1.bffffffffffffp-1` instead of the exact value `0.875 == 0x1.cp-1`. (And I have only tested on macOS so it may be different on other platforms.)

I **think** it's possible to get exact output if I hold off on doing divisions until "the end" but right now the code does "early" divisions:

- When computing line `ax + by + c = 0` (as part of normalization process setting `a^2 + b^2 = 1`)
- As part of converting `d(s)` into a Bézier curve by setting `x_j = j / m` (for the degree `m`)